modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/DataStreamAutoshardingIT.java

```diff
@@ -566,4 +586,44 @@ private static void mockStatsForIndex(
             }
         }
     }
+
+    private static void resetTelemetry() {
+        for (PluginsService pluginsService : internalCluster().getInstances(PluginsService.class)) {
+            final TestTelemetryPlugin telemetryPlugin = pluginsService.filterPlugins(TestTelemetryPlugin.class).findFirst().orElseThrow();
+            telemetryPlugin.resetMeter();
+        }
+    }
+
+    private static void assertTelemetry(String expectedEmittedMetric) {
+        Map<String, List<Measurement>> measurements = new HashMap<>();
+        for (PluginsService pluginsService : internalCluster().getInstances(PluginsService.class)) {
+            final TestTelemetryPlugin telemetryPlugin = pluginsService.filterPlugins(TestTelemetryPlugin.class).findFirst().orElseThrow();
+
+            telemetryPlugin.collect();
+
+            List<String> autoShardingMetrics = telemetryPlugin.getRegisteredMetrics(InstrumentType.LONG_COUNTER)
+                .stream()
+                .filter(metric -> metric.startsWith("es.auto_sharding."))
+                .sorted()
+                .toList();
+
+            assertEquals(autoShardingMetrics, MetadataRolloverService.AUTO_SHARDING_METRIC_NAMES.values().stream().sorted().toList());
+
+            for (String metricName : MetadataRolloverService.AUTO_SHARDING_METRIC_NAMES.values()) {
+                measurements.computeIfAbsent(metricName, n -> new ArrayList<>())
+                    .addAll(telemetryPlugin.getLongCounterMeasurement(metricName));
+            }
+        }
+
+        // assert other metrics not emitted
+        MetadataRolloverService.AUTO_SHARDING_METRIC_NAMES.values()
+            .stream()
+            .filter(metric -> metric.equals(expectedEmittedMetric) == false)
+            .forEach(metric -> assertThat(measurements.get(metric), empty()));
```

Comment:
I had a failure on this line in the CI build, which I was unable to reproduce locally. This means some metric that should not have been emitted, was emitted. I added the first call to `resetTelemetry` on line 126, as I thought that perhaps the test were run out of order and somehow the same telemetry plugin was being used in another test causing the measurement to still be in the recorder (though this doesn't seem likely). Any ideas what could have caused this?